crates/vhost/src/vhost_user/message.rs

```diff
@@ -492,8 +506,26 @@ pub struct VhostUserMemoryRegion {
     pub user_addr: u64,
     /// Offset where region starts in the mapped memory.
     pub mmap_offset: u64,
+
+    #[cfg(feature = "xen")]
+    /// Xen specific flags.
+    pub xen_mmap_flags: u32,
+
+    #[cfg(feature = "xen")]
+    /// Xen specific data.
+    pub xen_mmap_data: u32,
+}
+
+impl VhostUserMemoryRegion {
+    fn is_valid_common(&self) -> bool {
+        self.memory_size != 0
+            && self.guest_phys_addr.checked_add(self.memory_size).is_some()
+            && self.user_addr.checked_add(self.memory_size).is_some()
+            && self.mmap_offset.checked_add(self.memory_size).is_some()
+    }
 }
 
+#[cfg(not(feature = "xen"))]
```

Comment:
I'm a little concerned about this different implementation if `xen` is enabled or not.

Should we document that if the `xen` feature is enabled, then this crate doesn't work except with xen?

Reply:
Maybe we can introduce a trait with 2 implementations that can be used together, or this will never happen?

Reply:
Will not happen unless vm-memory is updated too.